### PR TITLE
Recipe for mgs-canopy

### PR DESCRIPTION
**Pull request by keuv-grvl**
- [x] I have read the guidelines above.
- [x] This PR adds a new recipe.
- [ ] This PR updates an existing recipe.
- [ ] This PR does something else (explain below).

The [mgs-canopy-algorithm](https://github.com/fplaza/mgs-canopy-algorithm) repo have a pretty well written Makefile for system-wide install, but it does not work within Conda. This is due to `boost` location, even by setting `BOOST_INCDIR` and others. So I compile "manually" the sources.

### CODE REVIEW

File: recipes/mgs-canopy/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{% set name = "mgs-canopy" %}
+{% set version = "1.0" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  git_rev: 00a60e0
+  git_url: https://github.com/fplaza/mgs-canopy-algorithm.git
```
Comment (bgruening) [inline recipes/mgs-canopy/meta.yaml]: please use a tarball and a checksum

File: recipes/mgs-canopy/build.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+cd src/
+
+echo "> Compiling sources"
+for F in $(find . -name "*.cpp") ; do
+	FF=${F%.*}
+	g++ -o $FF.o -fopenmp -c -Wall -Wextra -O3 -march=native -I./ -I${PREFIX}/include -L${PREFIX}/lib $FF.cpp
+done;
+
+echo "> Building target: cc.bin"
+g++ -o cc.bin -fopenmp -I${PREFIX}/include -L${PREFIX}/lib -lboost_program_options $(find . -name "*.o")
```
Comment (bgruening) [inline recipes/mgs-canopy/build.sh]: Why is it cc.bin? Maybe a more useful name or is this intended?

Comment (keuv-grvl) [inline recipes/mgs-canopy/build.sh]: This is the default name in the [Makefile](https://github.com/fplaza/mgs-canopy-algorithm/blob/master/src/Makefile#L33). I supposed people who are used to this software would look for `cc.bin`.
I can create a link to have `cc.bin` and `canopy` for example.

Comment (bgruening) [inline recipes/mgs-canopy/build.sh]: @keuv-grvl please do!

**Comment by keuv-grvl** [comment]
@bgruening ping :)

File: recipes/mgs-canopy/meta.yaml (hunk 1 above)
Comment (keuv-grvl) [inline recipes/mgs-canopy/meta.yaml]: Unfortunately, there is no proper tarball (`<name>-<version>.tar.gz`) available in the Github repo

Comment (druvus) [inline recipes/mgs-canopy/meta.yaml]: You may use `https://github.com/fplaza/mgs-canopy-algorithm/archive/00a60e0.tar.gz`.

**Comment by bebatut** [comment]
@keuv-grvl Are you done with this recipe?

**Comment by keuv-grvl** [comment]
@bebatut yes, I am

**Comment by SilasK** [comment]
Hallo I installed the canopy algorithm vio bioconda and got the error:
`canopy --help`
`Illegal instruction`

can anybody help me?

**Comment by mbargull** [comment]
Hi @SilasK: I opened #6929 which might fix your issue once it is merged.

**Comment by SilasK** [comment]
Is it already possible to test it beforehand?

**Comment by mbargull** [comment]
Only if you'd compile it on one machine and test it on another one that only supports a true subset of CPU instructions of those available on the compiling machine.. I'd suppose this would require more effort than waiting for 30-60 minutes until a new build is available on the channel 😉

**Comment by mbargull** [comment]
BTW: I could easily reproduce the error you got since the machine I'm working on is plenty old, too :D

**Comment by mbargull** [comment]
@SilasK: The new build works fine for me when I run `canopy --help`.
(Please make sure you install the latest build with `conda install canopy=1.0=1` as there's currently an issue with Conda which likely picks up the older build if you only specify `conda install canopy` or `conda install canopy=1.0`)

**Comment by SilasK** [comment]
Thank you very much, seems to work. So you need 30 min to create a bioconda recipie? 

 

Kind regards

Silas

**Comment by mbargull** [comment]
For the build itself, yes, take 20-30 minutes as a minimum time, maybe. A recipe is actually usually build twice, once when it is still a pull request and then when it is merged into the master branch from which the packages for the Bioconda channel are created from. Depending on the recipes complexity and number of created packages (e.g., for different Python versions), those builds can take some more time.
For the issue at hand you were lucky, in that the cause was very obvious, so that it could be fixed quickly and (against the good practice I usually try to adhere to) I didn't wait for someone to review the changes :wink:.
See the paper on Bioconda at https://www.biorxiv.org/content/early/2017/10/21/207092: 
> The  usual  turnaround  time  of  above  workflow  is  fast  (Fig. 2d).   61%  of  the pull requests are merged within 5 hours.  Of those, 36% are even merged within 1 hour.  Only 18% of the pull requests need more than a day.  Hence, publishing software in Bioconda or updating already existing packages can be accomplished typically within minutes to a few hours.

**Pull request merged**